drivers/ethernet/eth_xlnx_gem.c

```diff
@@ -654,6 +657,43 @@ static enum ethernet_hw_caps eth_xlnx_gem_get_capabilities(
 	return caps;
 }
 
+/**
+ * @brief GEM config request function
+ * Fills out the config struct with more detailed
+ * information regaring the RX/TX HW checksum 
+ * offloading capabilities of the device.
+ *
+ * @param dev Pointer to the device data
+ * @param type Enum to select the config type
+ * @retval config Pointer to the config data
+ * @return 0 if the requested config type is supported
+ */
+static int eth_xlnx_gem_get_config(const struct device *dev,
+	enum ethernet_config_type type,
+	struct ethernet_config *config)
+{
+	int ret = -1;
+	const struct eth_xlnx_gem_dev_cfg *dev_conf = dev->config;
+
+	if(type == ETHERNET_CONFIG_TYPE_RX_CHECKSUM_SUPPORT)
+	{
+		if (dev_conf->enable_rx_chksum_offload) {
+			config->chksum_support = NET_IF_CHECKSUM_IPV4_HEADER | NET_IF_CHECKSUM_IPV4_TCP | NET_IF_CHECKSUM_IPV4_UDP | NET_IF_CHECKSUM_IPV6_HEADER | NET_IF_CHECKSUM_IPV6_TCP | NET_IF_CHECKSUM_IPV6_UDP;
+		}
+		ret = 0;
+	}
+
+	if(type == ETHERNET_CONFIG_TYPE_TX_CHECKSUM_SUPPORT)
+	{
+		if (dev_conf->enable_tx_chksum_offload) {
+			config->chksum_support = NET_IF_CHECKSUM_IPV4_HEADER | NET_IF_CHECKSUM_IPV4_TCP | NET_IF_CHECKSUM_IPV4_UDP | NET_IF_CHECKSUM_IPV6_HEADER | NET_IF_CHECKSUM_IPV6_TCP | NET_IF_CHECKSUM_IPV6_UDP;
```

Comment:
config->chksum_support is of type enum ethernet_checksum_support, whose values are prefixed ETHERNET_CHECKSUM_SUPPORT_... These values should be used, even though they just redirect to NET_IF_CHECKSUM_...